app/controllers/transcriptions_controller.rb

```diff
@@ -19,17 +21,49 @@ def update
     raise ActionController::BadRequest if type_invalid?
     raise ActionController::BadRequest unless whitelisted_attributes?
 
-    if approve?
+    if approving?
       authorize @transcription, :approve?
     else
       authorize @transcription
     end
 
     update_attrs['updated_by'] = current_user.login
     @transcription.update!(update_attrs)
+
+    if @transcription.status_previously_changed?
+      if approving?
+        upload_files_to_storage
+      else
+        remove_files_from_storage
+      end
+    end
+
     render jsonapi: @transcription
   end
 
+  def export
+    @transcription = Transcription.find(params[:id])
+    authorize @transcription
+
+    data_storage = DataExports::DataStorage.new
+    data_storage.zip_transcription_files(@transcription) do |zip_file|
+      send_export_file zip_file
+    end
+  end
+
+  def export_group
+    @transcriptions = policy_scope([:export, Transcription]).where(group_id: params[:group_id])
```

Comment:
Per a conversation we just had: This will return all transcriptions that share a group id across all workflows and projects that a user has access to. 

But, if `export_group` requires a project_id param, then rather than muck about trying to satisfy `verify_policy_scoped` and all that above, you can just `authorize Project.find(params[:project_id])` and then scope your transcriptions lookup to the project/workflow/group you're trying to export.

Reply:
Now filtering by `group_id` and `workflow_id` within `export_group` 🎉